File: server/src/main/java/com/example/jira/model/Issue.java
package com.example.jira.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "issues")
public class Issue {

    @Id
    @JsonProperty("_id")
    private String id;

    private String title;
    private String description = "";
    private String type = "TASK";
    private String priority = "MEDIUM";
    private String status = "TODO";

    @Field("projectId")
    private Object projectId;

    @Field("assignee")
    private Object assignee;

    private Date dueDate;
    private Date createdAt = new Date();
    private Date updatedAt = new Date();
}

File: client/lib/models/issue.ts
export type IssueType = "TASK" | "BUG" | "STORY";

export type IssueStatus =
    | "TODO"
    | "IN_PROGRESS"
    | "IN_REVIEW"
    | "DONE";

    export type IssuePriority = "LOW" | "MEDIUM" | "HIGH";

    export interface Issue {
    id: string;
    key: string;

    title: string;
    description: string;

    type: IssueType;
    status: IssueStatus;
    priority: IssuePriority;

    assignee: string;
    dueDate?: string;

    createdAt?: string;
    updatedAt?: string;
}